shotover-proxy/src/transforms/cassandra/connection.rs

```diff
@@ -158,3 +157,46 @@ async fn rx_process<C: CodecReadHalf, T: AsyncRead>(
     }
     Ok(())
 }
+
+pub async fn receive(
+    failed_requests: &metrics::Counter,
+    mut results: FuturesOrdered<oneshot::Receiver<Response>>,
+) -> Result<Messages> {
+    let expected_size = results.len();
+    let mut responses = Vec::with_capacity(expected_size);
+    loop {
+        match timeout(Duration::from_secs(5), results.next()).await {
```

Comment:
This timeout should probably be configurable

Reply:
Why does cassandra even have a timeout here?
Redis doesnt seem to have one?

Should we just remove the timeout from cassandra to match redis?
Or should we add configurable timeouts to redis as well?

I pulled this logic into a prereq PR, so the decision we make can go into there: https://github.com/shotover/shotover-proxy/pull/727

Reply:
hmm we could either let the client handle the timeout. E.g. if the client ends up timing out... all this will get dropped anyway right? If the client hangs on waiting for a connection, so do we?

Reply:
A well behaving client should close the connection when it times out.
Shotover would definitely terminate the transform if its waiting for the client to send a message and the client closes the connection.
But I think thats not true if the transform is stuck waiting on something, in this case for a response from the server...

Reply:
Having thought about it.
If we want to handle the server suddenly never talking to us again, then we need to use a timeout like we already have for cassandra.
Redis currently doesnt do that, so I'll raise an issue to get that implemented.
For cassandra I'll make the timeout configurable and do that in the other PR.